			gcc[i386]						gcc[i686]
  -falign-functions=                    4                     |   -falign-functions=                    16
  -falign-jumps=                        4                     |   -falign-jumps=                        16:11:8
  -falign-labels=                                             |   -falign-labels=                       0:0:8
  -falign-loops=                        4                     |   -falign-loops=                        16
  -fschedule-insns2                     [disabled]            |   -fschedule-insns2                     [enabled]
  -falign-functions=                    4                     |   -falign-functions=                    16
  -falign-jumps=                        4                     |   -falign-jumps=                        16:11:8
  -falign-labels=                                             |   -falign-labels=                       0:0:8
  -falign-loops=                        4                     |   -falign-loops=                        16
  -march=                               i386                  |   -march=                               i686
  -mbranch-cost=<0,5>                   1                     |   -mbranch-cost=<0,5>                   2
  -mtune=                               i386                  |   -mtune=                               i686
  -falign-functions=                    4                     |   -falign-functions=                    16
  -falign-jumps=                        4                     |   -falign-jumps=                        16:11:8
  -falign-labels=                                             |   -falign-labels=                       0:0:8
  -falign-loops=                        4                     |   -falign-loops=                        16
  -fschedule-insns2                     [disabled]            |   -fschedule-insns2                     [enabled]
  -maccumulate-outgoing-args            [disabled]            |   -maccumulate-outgoing-args            [enabled]
  -march=                               i386                  |   -march=                               i686
  -mbranch-cost=<0,5>                   1                     |   -mbranch-cost=<0,5>                   2
  -mtune=                               i386                  |   -mtune=                               i686

		gcc[i386]						gcc[k6]
  -falign-functions=                    4                     |   -falign-functions=                    32
  -falign-jumps=                        4                     |   -falign-jumps=                        32:8:8
  -falign-labels=                                             |   -falign-labels=                       0:0:8
  -falign-loops=                        4                     |   -falign-loops=                        32:8:8
  -fschedule-insns2                     [disabled]            |   -fschedule-insns2                     [enabled]
  -falign-functions=                    4                     |   -falign-functions=                    32
  -falign-jumps=                        4                     |   -falign-jumps=                        32:8:8
  -falign-labels=                                             |   -falign-labels=                       0:0:8
  -falign-loops=                        4                     |   -falign-loops=                        32:8:8
  -march=                               i386                  |   -march=                               k6
  -mtune=                               i386                  |   -mtune=                               k6
  -falign-functions=                    4                     |   -falign-functions=                    32
  -falign-jumps=                        4                     |   -falign-jumps=                        32:8:8
  -falign-labels=                                             |   -falign-labels=                       0:0:8
  -falign-loops=                        4                     |   -falign-loops=                        32:8:8
  -fschedule-insns2                     [disabled]            |   -fschedule-insns2                     [enabled]
  -march=                               i386                  |   -march=                               k6
  -mmmx                                 [disabled]            |   -mmmx                                 [enabled]
  -mtune=                               i386                  |   -mtune=                               k6


---

Index: linux/arch/x86/Makefile
===================================================================
--- a/arch/x86/Makefile
+++ b/arch/x86/Makefile
@@ -22,7 +22,7 @@ else ifneq ($(call cc-option, -mstack-al
       cc_stack_align8 := -mstack-alignment=8
 endif
 
-# How to compile the 16-bit code.  Note we always compile for -march=i386;
+# How to compile the 16-bit code.  Note we always compile for -march=i686;
 # that way we can complain to the user if the CPU is insufficient.
 #
 # The -m16 option is supported by GCC >= 4.9 and clang >= 3.5. For
@@ -32,7 +32,7 @@ CODE16GCC_CFLAGS := -m32 -Wa,$(srctree)/
 M16_CFLAGS	 := $(call cc-option, -m16, $(CODE16GCC_CFLAGS))
 
 REALMODE_CFLAGS	:= $(M16_CFLAGS) -g -Os -DDISABLE_BRANCH_PROFILING \
-		   -Wall -Wstrict-prototypes -march=i386 -mregparm=3 \
+		   -Wall -Wstrict-prototypes -march=i686 -mregparm=3 \
 		   -fno-strict-aliasing -fomit-frame-pointer -fno-pic \
 		   -mno-mmx -mno-sse
 
@@ -63,8 +63,8 @@ KBUILD_CFLAGS += $(call cc-option,-mno-a
 
 ifeq ($(CONFIG_X86_32),y)
         BITS := 32
-        UTS_MACHINE := i386
-        CHECKFLAGS += -D__i386__
+        UTS_MACHINE := i686
+        CHECKFLAGS += -D__i686__
 
         biarch := $(call cc-option,-m32)
         KBUILD_AFLAGS += $(biarch)
@@ -97,10 +97,10 @@ else
         KBUILD_CFLAGS += -m64
 
         # Align jump targets to 1 byte, not the default 16 bytes:
-        KBUILD_CFLAGS += $(call cc-option,-falign-jumps=1)
+        KBUILD_CFLAGS += $(call cc-option,-falign-jumps=32:21:16)
 
         # Pack loops tightly as well:
-        KBUILD_CFLAGS += $(call cc-option,-falign-loops=1)
+        KBUILD_CFLAGS += $(call cc-option,-falign-loops=32:21:16)
 
         # Don't autogenerate traditional x87 instructions
         KBUILD_CFLAGS += $(call cc-option,-mno-80387)
#--- a/init/Kconfig
#+++ b/init/Kconfig
#@@ -1272,7 +1272,6 @@ config CC_OPTIMIZE_FOR_PERFORMANCE
# 
# config CC_OPTIMIZE_FOR_PERFORMANCE_O3
# 	bool "Optimize more for performance (-O3)"
#-	depends on ARC
# 	help
# 	  Choosing this option will pass "-O3" to your compiler to optimize
# 	  the kernel yet more for performance.
